android/key.go

```diff
@@ -0,0 +1,137 @@
+// Package android contains helper functions and types for Android.
+package android
+
+import "encoding/asn1"
+
+// AuthorizationList is the keymaster authorization list.
+type AuthorizationList struct {
+	Purpose                     KeyMasterPurposeSet `asn1:"tag:1,explicit,set,optional"`
+	Algorithm                   int                 `asn1:"tag:2,explicit,optional"`
+	KeySize                     int                 `asn1:"tag:3,explicit,optional"`
+	Digest                      []int               `asn1:"tag:5,explicit,set,optional"`
+	Padding                     []int               `asn1:"tag:6,explicit,set,optional"`
+	EcCurve                     int                 `asn1:"tag:10,explicit,optional"`
+	RsaPublicExponent           int                 `asn1:"tag:200,explicit,optional"`
+	RollbackResistance          asn1.Flag           `asn1:"tag:303,explicit,optional"`
+	ActiveDateTime              int                 `asn1:"tag:400,explicit,optional"`
+	OriginationExpireDateTime   int                 `asn1:"tag:401,explicit,optional"`
+	UsageExpireDateTime         int                 `asn1:"tag:402,explicit,optional"`
+	NoAuthRequired              asn1.Flag           `asn1:"tag:503,explicit,optional"`
+	UserAuthType                int                 `asn1:"tag:504,explicit,optional"`
+	AuthTimeout                 int                 `asn1:"tag:505,explicit,optional"`
+	AllowWhileOnBody            asn1.Flag           `asn1:"tag:506,explicit,optional"`
+	TrustedUserPresenceRequired asn1.Flag           `asn1:"tag:507,explicit,optional"`
+	TrustedConfirmationRequired asn1.Flag           `asn1:"tag:508,explicit,optional"`
+	UnlockedDeviceRequired      asn1.Flag           `asn1:"tag:509,explicit,optional"`
+	AllApplications             asn1.Flag           `asn1:"tag:600,explicit,optional"`
+	ApplicationID               asn1.Flag           `asn1:"tag:601,explicit,optional"`
+	CreationDateTime            int                 `asn1:"tag:701,explicit,optional"`
+	Origin                      KeyOrigin           `asn1:"tag:702,explicit,optional"`
+	RootOfTrust                 RootOfTrust         `asn1:"tag:704,explicit,optional"`
+	OsVersion                   int                 `asn1:"tag:705,explicit,optional"`
+	OsPatchLevel                int                 `asn1:"tag:706,explicit,optional"`
+	AttestationApplicationID    []byte              `asn1:"tag:709,explicit,optional"`
+	AttestationIDBrand          []byte              `asn1:"tag:710,explicit,optional"`
+	AttestationIDDevice         []byte              `asn1:"tag:711,explicit,optional"`
+	AttestationIDProduct        []byte              `asn1:"tag:712,explicit,optional"`
+	AttestationIDSerial         []byte              `asn1:"tag:713,explicit,optional"`
+	AttestationIDImei           []byte              `asn1:"tag:714,explicit,optional"`
```

Comment:
Flagging, but not sure how best to handle "IMEI" and "MEID" which are typically capitalized (similar to ID).

Though, in this particular case `AttestationIDImei` is much more readable than `AttestationIDIMEI`🤷‍♂️